k9mail-library/src/test/java/com/fsck/k9/mail/filter/FixedLengthInputStreamTest.java

```diff
@@ -0,0 +1,230 @@
+
+        final int INDEX = 888;
+        final int ARRAY_LEN = 10000;
+
+        fixedLengthInputStream.read(new byte[ARRAY_LEN], 0, INDEX);
+
+        int b = fixedLengthInputStream.read();
+        Assert.assertEquals(b, string.charAt(INDEX));
+    }
+
+    @Test
+    public void testMultiRead() throws IOException {
+        fixedLengthInputStream = new FixedLengthInputStream(new ByteArrayInputStream(string.getBytes()), LENGTH_LIMIT1);
+
+        final int LENGTH1 = 2999, LENGTH2 = 3001;   // LENGTH1 + LENGTH2 > LENGTH_LIMIT1
+
+        byte[] bytes = new byte[LENGTH_LIMIT1];
+        fixedLengthInputStream.read(bytes, 0, LENGTH1);
+        fixedLengthInputStream.read(bytes, LENGTH1, LENGTH2);
+        final byte[] bytes1 = string.substring(0, LENGTH_LIMIT1).getBytes();
+        Assert.assertArrayEquals(bytes, bytes1);
+    }
+
+
+    
+    // Test the cases that fixed length is larger than the length of bytes array
+
+    @Test
+    public void testLongInput2() throws IOException {
+        fixedLengthInputStream = new FixedLengthInputStream(new ByteArrayInputStream(string.getBytes()), LENGTH_LIMIT2);
+
+        final int LENGTH = 15000;   // LENGTH_LIMIT2 > LENGTH > STR_LENGTH
+        byte[] bytes = new byte[LENGTH_LIMIT2];
+
+        Assert.assertEquals(fixedLengthInputStream.read(bytes, 0, LENGTH), STR_LENGTH);
+        Assert.assertArrayEquals(bytes, concatByteArrays(string.getBytes(), new byte[LENGTH_LIMIT2 - STR_LENGTH]));
+    }
+
+    @Test
+    public void testLengthLimit2() throws IOException {
+        fixedLengthInputStream = new FixedLengthInputStream(new ByteArrayInputStream(string.getBytes()), LENGTH_LIMIT2);
+
+        fixedLengthInputStream.read(new byte[LENGTH_LIMIT2], 0, LENGTH_LIMIT2);
+
+        int b = fixedLengthInputStream.read();
+        Assert.assertEquals(b, -1);
+    }
+
+    @Test
+    public void testMultiRead2() throws IOException {
+        fixedLengthInputStream = new FixedLengthInputStream(new ByteArrayInputStream(string.getBytes()), LENGTH_LIMIT2);
+
+        final int LENGTH1 = 8999, LENGTH2 = 3001;       // LENGTH1 + LENGTH2 > STR_LENGTH > LENGTH1
+
+        byte[] bytes = new byte[LENGTH_LIMIT2];
+        fixedLengthInputStream.read(bytes, 0, LENGTH1);
+        fixedLengthInputStream.read(bytes, LENGTH1, LENGTH2);
+        final byte[] bytes1 = concatByteArrays(string.getBytes(), new byte[LENGTH_LIMIT2 - STR_LENGTH]);
+        Assert.assertArrayEquals(bytes, bytes1);
+    }
+
+    private byte[] concatByteArrays(byte[]... byteArrays) {
+        int totalLen = 0;
+        for (byte[] bytes : byteArrays) {
+            totalLen += bytes.length;
+        }
+
+        byte[] res = new byte[totalLen];
+
+        int start = 0;
+        for (byte[] bytes : byteArrays) {
+
+            System.arraycopy(bytes, 0, res, start, bytes.length);
+
+            start += bytes.length;
+        }
+
+        return res;
+    }
+}
```

Comment:
There are no tests for the methods

* `available()`
* `read()`
* `read(byte)`
* `skip(long)`
* `skipRemaining()`